app/components/input_field_component.rb

```diff
@@ -0,0 +1,9 @@
+class InputFieldComponent < ViewComponent::Base
+    def initialize(label: , name:, type: "text", width: "100%", margin_bottom: "0px")
+        @label = label
+        @name = name
+        @type = type
+        @width = width

```

Comment:
don't put styles as params;  make it by default have width 100%; then the parent container using it , will fix the a width